Dynavity/Dynavity/persistence/data-mapping/OnlineCanvasDTO.swift

```diff
@@ -17,8 +17,8 @@ struct OnlineCanvasDTO: Mappable {
     }
 
     func toModel() -> OnlineCanvas {
-        OnlineCanvas(ownerId: ownerId, canvas: canvas.toModel())
+        let model = OnlineCanvas(ownerId: ownerId, canvas: canvas.toModel())
+        CanvasRepository().addCloudChangeListeners(model: model)
```

Comment:
Would it be better to have a wrapper method in `CanvasRepository`  that calls `toModel` of the `OnlineCanvasDTO`, then adds the listeners? This is to remove the dependency from DTO to Repository.
Or maybe `addCloudChangeListeners` in `CanvasRepository` can be another method that takes in an `OnlineCanvasDTO`, converts it to a model, adds the listeners, then returns the `OnlineCanvas`?

Reply:
`addCloudChangeListeners` is not exclusively used in `OnlineCanvasDTO`. When a canvas is published, a new `OnlineCanvas` is made, and that will also require listeners to be attached.

I do admit however that there is a bit of duplication going on, since in principle we want every `OnlineCanvas` to be listening to the cloud (or whatever `OnlineStorageManager` the `CanvasRepository` has). I did consider performing `addCloudChangeListeners` in the OnlineCanvas initialiser, but that might open another bunch of problems.